### ISSUE TITLE
PodDisruptionBudgets don't work correctly 

### ISSUE THREAD

**Issue by mrkm4ntr** [elastic/cloud-on-k8s#1703]
## Bug Report

For PDB to work correctly, pods selected by PDB must have a specific controller reference.
The controller of Pods have to be Deployment, ReplicaSet, StatefulSet, RelicationController or CRD with scale subresource.
Refer following code.
https://github.com/kubernetes/kubernetes/blob/6348200c92dec8848e55552f3e8039b3da95bd91/pkg/controller/disruption/disruption.go#L178-L181

But we cannot add scale subresource into current Elasticsearch CRD, because it may have multiple `nodeCount`s. We have to rethink the definition of Elasticsearch CRD.

**What did you do?**
Created quickstart ElasticSearch.

**What did you expect to see?**
`quickstart-es-default` PDB have 1 allowed disruption.

**What did you see instead? Under which circumstances?**
`quickstart-es-default` PDB have 0 allowed disruption, and following warning.

```
Name:             quickstart-es-default
Namespace:        elastic-system
Max unavailable:  1
Selector:         elasticsearch.k8s.elastic.co/cluster-name=quickstart
Status:
    Allowed disruptions:  0
    Current:              0
    Desired:              0
    Total:                0
Events:
  Type     Reason         Age                   From               Message
  ----     ------         ----                  ----               -------
  Warning  NoControllers  52m (x62 over 77m)    controllermanager  found no controllers for pod "quickstart-es-6b9d28shzw"
  Warning  NoControllers  47m (x2 over 51m)     controllermanager  found no controllers for pod "quickstart-es-gg9vdmzgzt"
  Warning  NoControllers  42m (x3 over 43m)     controllermanager  found no controllers for pod "quickstart-es-hzrnqhbfns"
  Warning  NoControllers  12m (x18 over 44m)    controllermanager  found no controllers for pod "quickstart-es-pr7n5hj5f5"
  Warning  NoControllers  2m11s (x69 over 44m)  controllermanager  found no controllers for pod "quickstart-es-7frw9gsk2j"
```

**Environment**

* ECK version:

	insert version or git commit id here

* Kubernetes information:

	insert any information about your Kubernetes environment that could help us:
	* On premise ?
	* Cloud: GKE / EKS / AKS ?
	* Kubernetes distribution: Openshift / Rancher / PKS ?
	
	for each of them please give us the version you are using
```
$ kubectl version
```

* Resource definition:
```
if relevant insert the resource definition
```

* Logs:
```
insert operator logs or any relevant message to the issue here
```

**Comment by thbkrkr** [comment]
Related to #1455.

**Comment by pebrc** [comment]
I think this problem is addressed by the move to stateful sets see #1173 If I understand correctly we now have an owner reference that supports the virtual scale sub resource. 

```yaml
 ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: StatefulSet
    name: hello-eck-es-default
    uid: 478ef250-d877-11e9-a936-42010a840103
```
So I think we can close this. Please reopen if I am missing something.
